luma_core/src/ipc.rs

```diff
@@ -1,26 +1,26 @@
-use core::ptr::from_exposed_addr_mut;
+use core::ptr;
 
 //bits 0..=31 = physical address of ipc request
-const HW_IPC_PPCMSG: usize = 0xCD00_0000usize; //from_exposed_addr_mut(0xCD00_0000);
+const HW_IPC_PPCMSG: *mut u32 = ptr::without_provenance_mut(0xCD00_0000);
 
 //bit 0 = X1 | Execute IPC request
 //bit 1 = Y2 | Acknowledge IPC request
 //bit 2 = Y1 | IPC request reply available
 //bit 3 = X2 | Relaunch IPC
 //bit 4 = IY1 | IPC request reply send out IPC interrupt
 //bit 5 = IY2 | IPC request acknowledge sends out IPC interrupt
-const HW_IPC_PPCCTRL: usize = 0xCD00_0004usize; //from_exposed_addr_mut(0xCD00_0004);
+const HW_IPC_PPCCTRL: *mut u32 = ptr::without_provenance_mut(0xCD00_0004);
 
 //bits 0..=31 = physical address of ipc request
-const HW_IPC_ARMMSG: usize = 0xCD00_0008usize; //from_exposed_addr_mut(0xCD00_0008);
+const HW_IPC_ARMMSG: *mut u32 = ptr::without_provenance_mut(0xCD00_0008);
 
 //bit 0 = Y1 | IPC request reply available
 //bit 1 = X2 | Relauch IPC
 //bit 2 = X1 | Execute IPC request
 //bit 3 = Y2 | Acknowledge IPC request
 //bit 4 = IX1 | Execute ipc request send IPC interrupt
 //bit 5 = IX2 | Relaunch IPC sends IPC interrupt
-const HW_IPC_ARMCTRL: usize = 0xCD00_000Cusize; //from_exposed_addr_mut(0xCD00_000C);
+const HW_IPC_ARMCTRL: *mut u32 = ptr::without_provenance_mut(0xCD00_000C);
```

Comment:
we want `with_provenance_mut` here instead of `without_provenance_mut` otherwise writing to value is UB